include/megaapi.h

```diff
@@ -12529,6 +12684,103 @@ class MegaApi
+         * needed for a call to backgroundMediaUploadComplete.   The object must have been serialised
+         * before the app was unloaded.
+         *
+         * @return A pointer to an object that keeps some needed state through the process of
+         *         uploading a media file via iOS low power background uploads (or similar).
+         *         Caller takes ownership of the object.
+         */
+        MegaBackgroundMediaUpload* backgroundMediaUploadResume(const std::string* serialised);
+
+        /**
+         * @brief Request the URL suitable for uploading a media file.
+         *
+         * This function requests the URL needed for uploading the file.  The URL will need the urlSuffix
+         * from the MegaBackgroundMediaUpload::encryptFile to be appended before actually sending.
+         * The result of the request is signalled by the listener onRequestFinsish callback with TYPE_GET_BACKGROUND_UPLOAD_URL.
+         * Provided the error code is API_OK, the URL is available in the MegaBackgroundMediaUpload via its getUploadURL().
+         *
+         * Call this function just once (per file) to find out the URL to upload to, and upload all the pieces to the same
+         * URL.   If errors are encountered and the operation must be restarted from scratch, then a new URL should be requested.
+         * A new URL could specify a different upload server for example.
+         *
+         * @param fullFileSize The size of the file
+         * @param state A pointer to the MegaBackgroundMediaUpload object tracking this upload
+         * @param listener The MegaRequestListener to be called back with the result
+         */
+        void backgroundMediaUploadRequestUploadURL(int64_t fullFileSize, MegaBackgroundMediaUpload* state, MegaRequestListener *listener);
+
+        /**
+         * @brief Create the node after completing the background upload of the file.
+         *
+         * Call this function after completing the background upload of all the file data
+         * The node representing the file will be created in the cloud, with all the suitable
+         * attributes and file attributes attached.
+         * The result of the request is signalled by the listener onRequestFinsish callback with TYPE_COMPLETE_BACKGROUND_UPLOAD.
+         *
+         * @param state The MegaBackgroundMediaUpload object tracking this upload
+         * @param utf8Name The leaf name of the file, utf-8 encoded
+         * @param parent The folder node under which this new file should appear
+         * @param fingerprint  The fingerprint for the uploaded file (use MegaApi::getFingerprint to generate this)
+         * @param fingerprintoriginal If the file uploaded is modified from the original,
+         *        pass the fingerprint of the original file here, otherwise NULL.
+         * @param binaryUploadToken The N binary bytes of the token returned from the file upload (of the last portion). N=36 currently.
```

Comment:
About this one, I was concerned that it would be difficult to specify the encoding to use, and felt that straight binary would be the least difficult to get the client app to supply correctly.   We don't provide any binary-to-64 functions for the client to use (that I can see?), and I presume that is deliberate?  thanks

Reply:
Well, there are methods to convert handles from/to B64, but it's true that till now there was no need to provide converters for any size. Maybe it's time to do it?